### PR TITLE
C/cpp: Bubble Sort

### LINKED ISSUE #878: CPP: Bubble Sort
<hr>

## Description 📜

I want to create a documentation explaining bubble sort technique in C++.

<hr>

## Domain of Contribution 📊

- [x ] C/CPP

<hr>

### PR DESCRIPTION
**Pull request by Xyandraa9**
<hr>

## Description 📜

I added the documentation for Bubble sorting in Arrays.
Fixes #878

<hr>

## Type of change 📝

- [x] Doc#umentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/#CPP
<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Review by vasu-1 — APPROVED** [review]
Great job @Xyandraa9 
Merge this @Suvraneel 
And move it to data structure-> sorting 
folder

File: C_CPP/Arrays/BubbleSort.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+   
+   for(i = 0; i < 10; i++){
+    for(j = i+1; j < 10; j++){ // j starts from the next element for comparison.
+      if(a[j] < a[i]) { // swapping values here if previous value is less than the current value.
+         temp = a[i];
+         a[i] = a[j];
+         a[j] = temp;
+      }
+    }
+
```
Comment (deeqakkk) [inline C_CPP/Arrays/BubbleSort.md:30]: remove this extra line space

File: C_CPP/Arrays/BubbleSort.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  ```
+  
+   ## Output:
+   (Ascending order)
+   ![image](https://user-images.githubusercontent.com/75472177/134164877-095ced7e-ea29-4120-a30a-09fa33ce8425.png)
+   
+   (Descending order)
+   ![image](https://user-images.githubusercontent.com/75472177/134165046-c419a5c6-1f03-46f0-9c13-a419b40271ac.png)
+   
+## Time Complexity of Bubble Sorting:
```
Comment (deeqakkk) [inline C_CPP/Arrays/BubbleSort.md:57]: You can also add "stability" and "adaptivity" of this sorting algorithm.

File: C_CPP/Arrays/BubbleSort.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ Sum = (N-1) + (N-2) + (N-3) + .... + 2 + 1 
+ 
+ Sum = N(N-1) / 2 
+ 
+ 
+  = O(n^2) 
+  
+  
+Hence the worst-case time complexity is O(n^2). 
+
```
Comment (deeqakkk) [inline C_CPP/Arrays/BubbleSort.md:70]: Mention best-case time complexity as well.

**Review by deeqakkk — CHANGES_REQUESTED** [review]

File: C_CPP/Arrays/BubbleSort.md (hunk 3 above)
Comment (Xyandraa9) [reply C_CPP/Arrays/BubbleSort.md:68]: @deeqakkk thankyou for your suggestions, kindly review them and merge.

File: C_CPP/Arrays/BubbleSort.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+When the input array is already sorted, that becomes the best-case time complexity. Hence the best-case time complexity of bubble sort is O(n).
+  
+## Space Complexity of Bubble Sorting:
+This algorithm requires only one additional memory space for the temp variable while swapping. Hence the space complexity for bubble sort algorithm is O (1).
+
+
+## Stability of Bubble Sorting:
+
+Stability of a sorting algorithm means that- when two instances with equal values appear in the same order in the sorted output as they appear in the array as given by the user. Hence, Bubble Sort is stable.
```
Comment (Suvraneel) [inline C_CPP/Arrays/BubbleSort.md:77]: Add Reason of stability : Only adjacent element swaps occur

**Review by Suvraneel — CHANGES_REQUESTED** [review]
1. Move this to Data Structures > Sorting
2. Add a line to readme

File: C_CPP/Arrays/BubbleSort.md (hunk 4 above)
Comment (Xyandraa9) [reply C_CPP/Arrays/BubbleSort.md]: @Suvraneel , made the changes. Kindly review and merge.

**Review by deeqakkk — APPROVED** [review]
Good to be merged @Suvraneel .

**Comment by Suvraneel** [comment]
@Xyandraa9  Awesome ❤️ 
Btw I've relocated the folders a bit today... So shifted your's to the new Location
Feel free to check it out

**Comment by Xyandraa9** [comment]
@Suvraneel Thankyou very much, and yes I checked it out!

**Comment by msbuddhu** [comment]
@Suvraneel Why I have been unassigned? It has been only one day.

**Comment by vasu-1** [comment]
> @Suvraneel Why I have been unassigned? It has been only one day.

Only issues can be assigned not the Pr :)
- Create the Issue and assign to yourself (Must not be duplicate_)

**Pull request merged**